nri/container_linux.go

```diff
@@ -0,0 +1,35 @@
+//go:build linux
+// +build linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package nri
+
+import (
+	nri "github.com/containerd/nri/v2alpha1/pkg/adaptation"
+)
+
+func linuxContainerToNRI(ctr Container) *nri.LinuxContainer {
```

Comment:
odd to have the name prefix "linux" here

Reply:
Hmm... not sure what to do about them. OCI and CRI both have Linux- and Windows-specific fields. Those are not defined in a platform-specific way. Instead any structure which needs platform-specific fields is defined to contain 'the union' of all those fields. In NRI we follow the same pattern, although we only have/use the Linux-specific ones of any fields that might be platform-specific.

One way I can think of to avoid having this Linux- and later potentially Windows-prefixes, is to reorganize this code so that first we only fill in the platform-agnostic common bits and then we call something like `platformContainerToNRI(ctr, nriCtr)` to take care of the platform-specific bits. That function can then be defined separately for Linux and Windows to deal with the corresponding fields.

Personally I suspect it would make the code less and not more readable... but I might be wrong. Just let me know if you'd prefer it that way (or if you have some other suggestions) and I can make the change...